### PR TITLE
8344855: Remove calls to SecurityManager and doPrivileged  in HTTP related implementation classes in the sun.net and sun.net.www.http packages after JEP 486 integration

### PR DESCRIPTION
**Pull request by dfuch**
Some further cleaning in the legacy HTTP implementation. 
Usual removal of doPrivileged, GetPropertyAction, checkPermission, etc...

I also took the opportunity to also remove some constructors that were never called in the legacy HttpClient, and to fix some throws signatures.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344855](https://bugs.openjdk.org/browse/JDK-8344855): Remove calls to SecurityManager and doPrivileged  in HTTP related implementation classes in the sun.net and sun.net.www.http packages after JEP 486 integration (**Sub-task** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22321/head:pull/22321` \
`$ git checkout pull/22321`

Update a local copy of the PR: \
`$ git checkout pull/22321` \
`$ git pull https://git.openjdk.org/jdk.git pull/22321/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22321`

View PR using the GUI difftool: \
`$ git pr show -t 22321`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22321.diff">https://git.openjdk.org/jdk/pull/22321.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22321#issuecomment-2493716154)
</details>

### CODE REVIEW

File: src/java.base/share/classes/sun/net/www/http/KeepAliveStreamCleaner.java [hunk 1]
```diff
@@ -64,22 +62,10 @@ class KeepAliveStreamCleaner
...
     static {
         final String maxDataKey = "http.KeepAlive.remainingData";
-        @SuppressWarnings("removal")
-        int maxData = AccessController.doPrivileged(
-            new PrivilegedAction<Integer>() {
-                public Integer run() {
-                    return NetProperties.getInteger(maxDataKey, 512);
-                }}).intValue() * 1024;
-        MAX_DATA_REMAINING = maxData;
+        MAX_DATA_REMAINING = NetProperties.getInteger(maxDataKey, 512) * 1024;
```
Comment (vy) [inline src/java.base/share/classes/sun/net/www/http/KeepAliveStreamCleaner.java:65]: I'm don't know the context, but `Math.multiplyExact()` can be a safer alternative.

File: src/java.base/share/classes/sun/net/NetProperties.java [hunk 2]
```diff
@@ -39,17 +37,10 @@
...
 public class NetProperties {
     private static Properties props = new Properties();
     static {
-        AccessController.doPrivileged(
-            new PrivilegedAction<Void>() {
-                public Void run() {
-                    loadDefaultProperties();
-                    return null;
-                }});
+        loadDefaultProperties();
```
[the remark this answers is not in the document]
Comment (dfuch) [reply src/java.base/share/classes/sun/net/NetProperties.java:43]: Good idea.

File: src/java.base/share/classes/sun/net/www/http/KeepAliveStreamCleaner.java (hunk 1 above)
Comment (dfuch) [reply src/java.base/share/classes/sun/net/www/http/KeepAliveStreamCleaner.java:65]: What happens if we overflow is that we will not try to discard any data from the keep alive stream, exactly as if a negative or zero value had been provided. That said, the KeepAliveStream compares that with `expected`, which is a long, so `MAX_DATA_REMAINING` should really be a long, not an int, and given that the property value is an int, that would prevent any overflow. I might just do that.

**Review by jaikiran — APPROVED** [review]
This looks good to me.

**Pull request closed without merging**